ganttproject/src/net/sourceforge/ganttproject/UIFacadeImpl.java

```diff
@@ -247,6 +236,23 @@ public void changeValue(ChangeValueEvent event) {
     myOptions.setTitled(false);
 
     myLogoOption = new DefaultFileOption("ui.logo");
+    myLogoOption.addChangeValueListener(new ChangeValueListener(){
+      @Override
+      public void changeValue(ChangeValueEvent event) {
+
+        if (event.getOldValue()!=null && event.getOldValue().equals(event.getNewValue())){
+          Image newLogo =getLogo();
+
+
+          ((GanttTree2)myFallbackDelegate.getTaskTree()).
```

Comment:
I don't think that we want to expose that many internal fields and do all these casts to concrete classes and call chains. 
What about exposing LogoOption from UIFacade instead? Class `ChartTabContentPanel` has access to `UIFacade` instance, it will be able to get the logo option and add a listener.